tests/subsys/dfu/dfu_target_mcuboot/src/main.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <string.h>
+#include <zephyr/types.h>
+#include <stdbool.h>
+#include <ztest.h>
+#include <dfu_target_mcuboot.h>
+
+/* Create buffer which we will fill with strings to test with.
+ * This is needed since 'dfu_ctx_Mcuboot_set_b1_file` will modify its
+ * 'file' parameter, hence we can not provide it raw c strings as they
+ * are stored in read-only memory
+ */
+char buf[1024];
+
+#define S0_S1 "s0 s1"
+#define NO_SPACE "s0s1"
```

Comment:
what about a failure case which is over 500 in len? also one which contains a non-terminated cstring.